.circleci/config.yml

```diff
@@ -1,115 +1,163 @@
 version: 2.1
 orbs:
-  android: circleci/android@1.0
+  android: circleci/android@2.1.2
+  codecov: codecov/codecov@3.2.4
 
 # Workflows orchestrate a set of jobs to be run;
 workflows:
   version: 2
   build-test-deploy:
     jobs:
-      - validate-code
+      - validate-code:
+          gradle-cache-prefix: v1
+          build-cache-prefix: v1
+
       - build-and-unit-test:
+          gradle-cache-prefix: v1
+          build-cache-prefix: v1
           requires:
             - validate-code
+      
       - functional-test:
+          gradle-cache-prefix: v1
+          build-cache-prefix: v1
           requires:
             - validate-code
- 
+
 jobs:
   validate-code:
-    working_directory: ~/code
-    docker:
-      - image: circleci/android:api-29-node
-    environment:
-      JVM_OPTS: -Xmx3200m
+    parameters:
+      gradle-cache-prefix:
+        type: string
+        default: v1
+      build-cache-prefix:
+        type: string
+        default: v1
+
+    # List of available Android Docker images: https://circleci.com/developer/images/image/cimg/android#image-tags
+    executor:
+      name: android/android-machine
+      resource-class: large
+      tag: 2022.09.1
+
     steps:
       - checkout
 
+      - android/restore-gradle-cache:
+          cache-prefix: << parameters.gradle-cache-prefix >>
+      - android/restore-build-cache:
+          cache-prefix: << parameters.build-cache-prefix >>
+      
       - run:
           name: Check Code Format
           command: make format-check
 
+      - run:
+          name: Lint Code
+          command: make lint
+          
+      - android/save-gradle-cache:
+          cache-prefix: << parameters.gradle-cache-prefix >>
+      - android/save-build-cache:
+          cache-prefix: << parameters.build-cache-prefix >> 
+
+      # Store Lint report
+      - store_artifacts:
+          path: code/edgeconsent/build/reports     
+
   build-and-unit-test:
-    working_directory: ~/code
-    docker:
-      - image: circleci/android:api-29-node
-    environment:
-      JVM_OPTS: -Xmx3200m
+    parameters:
+      gradle-cache-prefix:
+        type: string
+        default: v1
+      build-cache-prefix:
+        type: string
+        default: v1
+
+    # List of available Android Docker images: https://circleci.com/developer/images/image/cimg/android#image-tags
+    executor:
+      name: android/android-machine
+      resource-class: large
+      tag: 2022.09.1
```

Comment:
Same comment as before, use the Android Docker image.

Reply:
I will test that in the Gradle PR.